### PR TITLE
http channel deadconn detection

### LINKED ISSUE #285: [bug] RPC api forwards http headers incorrectly
**What happened**:
When invoking RPC api using typescript sdk written in #262 ,the client got an error when parsing the response frame.
```
Error: 13 INTERNAL: Received RST_STREAM with code 1
```

image

**What you expected to happen**:
The typescript sdk invokes RPC api successfully

**How to reproduce it (as minimally and precisely as possible)**:
Follow the README doc in #262 

**Anything else we need to know?**:
There are two bugs:
1. We found that this bug was caused by directly forwarding the 'Content-Length' header to the client.
If we modify Layotto's code and make it not to forward this field, the ut will invoke it successfully
image

2. Sometimes a different error occurs:
```
Error: 14 UNAVAILABLE: io: read/write on closed pipe
```

image

### PR DESCRIPTION
**Pull request by MoonShining**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #285 

1. https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.6 clarified content-length header is optional, so just drop it
2. add http connection close check

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/grpc/api.go [hunk 1]
```diff
@@ -230,9 +230,10 @@ func (a *api) InvokeService(ctx context.Context, in *runtimev1pb.InvokeServiceRe
 	if resp.Header != nil {
 		header := metadata.Pairs()
 		for k, values := range resp.Header {
-			for _, v := range values {
-				header.Append(k, v)
+			if strings.EqualFold("content-length", k) {
+				continue
```
Comment (wenxuwan) [inline pkg/grpc/api.go:234]: attach issue here for user understand

File: components/rpc/invoker/mosn/channel/httpchannel.go [hunk 2]
```diff
@@ -70,36 +120,39 @@ func (h *httpChannel) Do(req *rpc.RPCRequest) (*rpc.RPCResponse, error) {
...
+	if err = conn.SetWriteDeadline(deadline); err != nil {
 		h.pool.Put(conn, true)
 		return nil, common.Error(common.UnavailebleCode, err.Error())
 	}
 
 	httpReq := h.constructReq(req)
 	defer fasthttp.ReleaseRequest(httpReq)
 
+	hstate := conn.state.(*hstate)
+	hstate.renew()
```
Comment (wenxuwan) [inline components/rpc/invoker/mosn/channel/httpchannel.go:132]: why renew every time? we can do it only once on stateFunc

File: pkg/grpc/api.go (hunk 1 above)
Comment (MoonShining) [reply pkg/grpc/api.go:235]: ok

File: components/rpc/invoker/mosn/channel/httpchannel.go (hunk 2 above)
Comment (MoonShining) [reply components/rpc/invoker/mosn/channel/httpchannel.go]: ok

**Review by wenxuwan — APPROVED** [review]
LGTM

**Review by wenxuwan — APPROVED** [review]
LGTM

**Pull request merged**